OSDOCS-2855 Installing on IBM Cloud IPI

mjpytlak opened the pull request:
This is a continuation of https://github.com/openshift/openshift-docs/pull/39767.

CP to 4.10

https://issues.redhat.com/browse/OSDOCS-2855

mjpytlak commented:
Hi @pamoedom I am taking over the IBM Cloud doc with Cody's departure. My latest commit includes a few late breaking edits from the team. Looking for one final QE ack. This is fairly time sensitive, as doc freeze is this Friday, 2/25. Thanks in advance.

pamoedom commented:
Hi @mjpytlak , thanks for taking over, one question, is the generated preview ok? I see the IBMCloud section referring IPI BM stuff which makes no sense, example:
![Screenshot_2022-02-23_21-50-53](https://user-images.githubusercontent.com/43780184/155406618-178f9c71-ce46-4883-9361-ee7ad5a683dc.png)
![Screenshot_2022-02-23_21-53-15](https://user-images.githubusercontent.com/43780184/155406640-d85b491f-15ac-475a-b3c4-e6556a415f40.png)

mjpytlak commented:
Yes. That through me off, as well, the first time I previewed the build. Per Cody during his handover.

"We need to retitle the current doc set we have on installing bare metal IPI clusters to IBM Cloud . Telco owns this, and I have a PR up to get approval from that team. John Wilkins, the tech writer responsible, has ACK’d via Slack. Looking for PM ACK in the PR https://github.com/openshift/openshift-docs/pull/39716"

I will follow up with the Teleco writer.

ahardin-rh commented on installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc:
```diff
@@ -0,0 +1,24 @@
+
+include::modules/alternatives-to-storing-admin-secrets-in-kube-system.adoc[leveloffset=+1]
+
+.Additional resources
```
[role="_additional-resources"]
[id="additional-resources_configuring-iam-ibm-cloud"]
== Additional resources

per updated guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

mjpytlak replied:
Confirmed with Katherine earlier that .Additional resources is still OK, but the role and anchor ID is requiered.

mjpytlak replied:
Done.

mjpytlak replied:
Done.

mjpytlak replied:
@ahardin-rh So based on the conversation in the peer review Slack channel, @kalexand-rh confirmed that using .`Additional resources` below a module is still legit.

ahardin-rh commented on installing/installing_ibm_cloud_public/configuring-iam-ibm-cloud.adoc:
```diff
@@ -0,0 +1,24 @@
+// Will need to revisit upgrade scenario for IBM Cloud; not needed until OCP 4.11. Tentative instructions have been added for reference later.
+
+[id="next-steps_configuring-iam-ibm-cloud"]
+== Next steps
```
[id="next-steps_configuring-iam-ibm-cloud"]
== Next steps

per updated guidelines: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#anchoring-prerequisites-additional-resources-and-next-steps-titles-in-assemblies

mjpytlak replied:
@ahardin-rh not sure if I need an update here. The heading already has an anchor ID that adheres to the standards.

ahardin-rh commented on installing/installing_ibm_cloud_public/installing-ibm-cloud-customizations.adoc:
```diff
@@ -0,0 +1,61 @@
+
+In {product-title} version {product-version}, you can install a customized cluster on infrastructure that the installation program provisions on IBM Cloud. To customize the installation, you modify parameters in the `install-config.yaml` file before you install the cluster.
+
+== Prerequisites
```
[id="prerequisites_installing-ibm-cloud-customizations"]
== Prerequisites

mjpytlak replied:
Done.

ahardin-rh commented on installing/installing_ibm_cloud_public/preparing-to-install-on-ibm-cloud.adoc:
```diff
@@ -0,0 +1,46 @@
+
+toc::[]
+
+[id="{context}-prerequisites"]
```
[id="prerequisites_preparing-to-install-on-ibm-cloud"]
== Prerequisites

ahardin-rh commented on modules/installation-cis-ibm-cloud.adoc:
```diff
@@ -0,0 +1,67 @@
+
+. Create a CIS instance to use with your cluster.
+
+.. Install the CIS plugin:
```
plug-in

mjpytlak replied:
Done.

ahardin-rh commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1036,6 +1044,40 @@ The GCP Compute Engine System service account email, like `<service_account_name
+
+|`platform.ibmcloud.dedicatedHosts.profile`
+|The new dedicated host to create. If you specify a value for `platform.ibmcloud.dedicatedHosts.name`, this parameter is not required.
+|Valid IBM Cloud dedicated host profile, such as `cx2-host-152x304`. ^2^
```
Looking at https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#alternative-footnote-styling-in-tables you might need brackets in the footnote callouts, but looks like it's rendering okay in the preview build. I would just double check this.

mjpytlak replied:
Updated to include `[]`

ahardin-rh commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -1036,6 +1044,40 @@ The GCP Compute Engine System service account email, like `<service_account_name
+|====
+[.small]
+--
+1. Whether you define an existing resource group, or if the installer creates one, determines how the resource group is treated when the cluster is uninstalled. If you define a resource group, the installer removes all of the installer provisioned resources, but leaves the resource group alone; if a resource group is created as part of the installation, the installer removes all of the installer  ...
```
installer-provisioned resources

mjpytlak replied:
Done.

ahardin-rh commented on modules/installation-ibm-cloud-config-yaml.adoc:
```diff
@@ -0,0 +1,106 @@
+endif::[]
+
+[id="installation-ibm-cloud-config-yaml_{context}"]
+= Sample customized `install-config.yaml` file for IBM Cloud
```
No markup in headings, so:
= Sample customized install-config.yaml file for IBM Cloud

mjpytlak replied:
Done.

ahardin-rh commented on modules/installation-ibm-cloud-config-yaml.adoc:
```diff
@@ -0,0 +1,106 @@
+[IMPORTANT]
+====
+The use of FIPS Validated / Modules in Process cryptographic libraries is only supported on {product-title} deployments on the `x86_64` architecture.
+====
```
can we replace the / with or?

mjpytlak replied:
Yep. Done.

ahardin-rh commented on modules/manually-maintained-credentials-upgrade.adoc:
```diff
@@ -61,6 +80,27 @@ It may take several minutes after adding the annotation for the upgradeable stat
+. Revoke the old service IDs and API Keys:
++
+[source,terminal]
++
```
You don't need the line break between [source,terminal] and the start of the  code block

mjpytlak replied:
Done.

mjpytlak commented on modules/installation-uninstall-clouds.adoc:
```diff
@@ -48,10 +48,49 @@ endif::gcp[]
 endif::ibm-cloud[]
 
 .Procedure
+ifdef::ibm-cloud[]
```
@cjschaef Please take a pass at this procedure. In particular, can you define what `<infrastructure_id>` is? Thanks in advance.

jeffnowicki commented on modules/installation-uninstall-clouds.adoc:
```diff
@@ -48,10 +48,49 @@ endif::gcp[]
+$ ibmcloud is volumes --resource-group-name <infrastructure_id>
+----
++
+For more information about listing volumes, see the link:https://cloud.ibm.com/docs/vpc?topic=vpc-infrastructure-cli-plugin-vpc-reference&interface=ui#volume-cli[IBM VPC CLI documentation].
```
(minor) use "IBM Cloud VPC CLI"

mjpytlak replied:
Done.

jeffnowicki commented on modules/installation-uninstall-clouds.adoc:
```diff
@@ -48,10 +48,49 @@ endif::gcp[]
+$ ibmcloud is volume-delete --force <volume_id>
+----
++
+For more information about deleting volumes, see the link:https://cloud.ibm.com/docs/vpc?topic=vpc-infrastructure-cli-plugin-vpc-reference&interface=ui#volume-delete[IBM VPC CLI documentation].
```
(minor) use "IBM Cloud VPC CLI"

mjpytlak replied:
Done.

jeffnowicki reviewed (approved):
I only have 2 minor comments.

mjpytlak commented:
@ahardin-rh Peer review comments addressed. Thank you.
@jeffnowicki Addressed your comments. Thank you.

@pamoedom Outside of your concern about the TOC, do you have additional feedback on the last round of edits? If not, please approve so I can merge this doc by COB 2/25. Thank you.

pamoedom commented:
/label qe-approved

ahardin-rh commented:
All latest updates LGTM!

sagidlow commented:
/cherrypick enterprise-4.10
